src/v/cloud_storage/segment_meta_cstore.h

```diff
@@ -90,6 +94,7 @@ class segment_meta_frame_const_iterator
     std::optional<decoder_t> _decoder{std::nullopt};
     uint32_t _pos{0};
     uint32_t _size{0};
+    value_t _frame_initial;
```

Comment:
nit: can you add a comment explaining what this is? Without too much context it reads a bit strange for this to not be related to _head.

Reply:
Will do. Basically _head is an intermediate buffer where new data is appended until there is enough to be compressed (16 elements). _frame_initial is the first datapoint of this frame. It could be read from the compressed buffer, but it's faster to cache it